Moving an upgrade between zone don't work

**Describe the bug**
With evidence of crime I tried to move an upgrade from enemy ground arena to my space arena but the engine give me this error : `AssertionError: Contract assertion failure: Attempting to remove card smuggling-compartment for controller Player({ key:'player2' }) from groundArena but it is in the arena for the opponent`

**To Reproduce**
Here an example :
```
describe('Hondo Ohnaka, Superfluous Swindler', function() {
    integration(function(contextRef) {
        it('Hondo Ohnaka\'s ability takes control of an non-pilot upgrade and attach it to a different unit', async function () {
            await contextRef.setupTestAsync({
                phase: 'action',
                player1: {
                    hand: ['independent-smuggler'],
                    groundArena: [
                        { card: 'atst', upgrades: ['smuggling-compartment'] },
                        { card: 'wampa', upgrades: ['shield'] }
                    ],
                    spaceArena: ['phoenix-squadron-awing']
                },
                player2: {
                    hand: ['evidence-of-the-crime'],
                    groundArena: ['hondo-ohnaka#superfluous-swindler'],
                    spaceArena: ['green-squadron-awing'],
                    leader: 'finn#this-is-a-rescue',
                    resources: 4
                }
            });
            const { context } = contextRef;

            context.player1.clickCard(context.independentSmuggler);
            context.player1.clickPrompt('Play Independent Smuggler with Piloting');
            context.player1.clickCard(context.atst);

            context.player2.clickCard(context.evidenceOfTheCrime);
            // context.player2.clickCard(context.p1Base);
            expect(context.player2).toBeAbleToSelectExactly([context.smugglingCompartment, context.shield, context.independentSmuggler]);
            context.player2.clickCard(context.smugglingCompartment);
            expect(context.player2).toBeAbleToSelectExactly([context.phoenixSquadronAwing, context.greenSquadronAwing, context.atst]);
            context.player2.clickCard(context.greenSquadronAwing);

            expect(context.greenSquadronAwing).toHaveExactUpgradeNames([context.smugglingCompartment.name]);

            context.player1.passAction();

            // assert we have control of upgrade with finn ability which target only upgrade we control
            context.player2.clickCard(context.finn);
            expect(context.player2).toBeAbleToSelectExactly([context.smugglingCompartment]);
            context.player2.clickCard(context.smugglingCompartment);

            expect(context.player1).toBeActivePlayer();
            expect(context.greenSquadronAwing).toHaveExactUpgradeNames(['shield']);
        });
    });
});
```

--- a/server/game/core/card/baseClasses/PlayableOrDeployableCard.ts
+++ b/server/game/core/card/baseClasses/PlayableOrDeployableCard.ts
@@ -245,12 +245,17 @@ export class PlayableOrDeployableCard extends Card implements IPlayableOrDeploya
             `Attempting to take control of card ${this.internalName} for player ${newController.name} in invalid zone: ${moveDestination}`
         );
 
-        // if we're changing controller and staying in the arena, just tell the arena to update our controller. no move needed
-        if (moveDestination === this.zoneName && (this.zone.name === ZoneName.GroundArena || this.zone.name === ZoneName.SpaceArena)) {
+        // if we're changing controller and staying in play, tell the arena to update our controller
+        if (this.zone.name === ZoneName.GroundArena || this.zone.name === ZoneName.SpaceArena) {
             this.zone.updateController(this);
 
-            // register this transition with the engine so it can do uniqueness check if needed
-            this.registerMove(this.zone.name);
+            // if we're staying in the same arena, no move needed
+            if (moveDestination === this.zoneName) {
+                // register this transition with the engine so it can do uniqueness check if needed
+                this.registerMove(this.zone.name);
+            } else {
+                this.moveTo(moveDestination);
+            }
         } else {
             this.moveTo(moveDestination);
         }

--- a/test/server/cards/02_SHD/events/EvidenceOfTheCrime.spec.ts
+++ b/test/server/cards/02_SHD/events/EvidenceOfTheCrime.spec.ts
@@ -7,11 +7,18 @@ describe('Evidence of the Crime', function() {
                     player1: {
                         hand: ['evidence-of-the-crime'],
                         groundArena: [{ card: 'boba-fett#disintegrator', upgrades: ['boba-fetts-armor'] }],
-                        spaceArena: [{ card: 'avenger#hunting-star-destroyer', exhausted: true, upgrades: ['frozen-in-carbonite'] }],
+                        spaceArena: [
+                            { card: 'avenger#hunting-star-destroyer', exhausted: true, upgrades: ['frozen-in-carbonite'] },
+                            'green-squadron-awing'
+                        ],
                         leader: { card: 'boba-fett#daimyo', deployed: true, upgrades: ['grievouss-wheel-bike'] },
                     },
                     player2: {
-                        groundArena: [{ card: 'atst', exhausted: true }, { card: 'isb-agent', upgrades: ['boba-fetts-armor'] }],
+                        groundArena: [
+                            { card: 'atst', exhausted: true },
+                            { card: 'isb-agent', upgrades: ['boba-fetts-armor'] },
+                            { card: 'frontier-atrt', upgrades: ['smuggling-compartment'] }
+                        ],
                         spaceArena: ['cartel-spacer'],
                         leader: { card: 'darth-vader#dark-lord-of-the-sith', deployed: true, upgrades: ['the-darksaber'] }
                     }
@@ -29,7 +36,12 @@ describe('Evidence of the Crime', function() {
 
                 // Move Boba Fett Armor from player 1 to player 2
                 context.player1.clickCard(context.evidenceOfTheCrime);
-                expect(context.player1).toBeAbleToSelectExactly([context.frozenInCarbonite, context.p1BobaFettArmor, context.p2BobaFettArmor]);
+                expect(context.player1).toBeAbleToSelectExactly([
+                    context.frozenInCarbonite,
+                    context.p1BobaFettArmor,
+                    context.p2BobaFettArmor,
+                    context.smugglingCompartment
+                ]);
 
                 context.player1.clickCard(context.p1BobaFettArmor);
                 expect(context.player1).toBeAbleToSelectExactly([context.bobaFettDisintegrator, context.bobaFettDaimyo, context.isbAgent, context.darthVader]);
@@ -43,10 +55,23 @@ describe('Evidence of the Crime', function() {
                 // Move Frozen in Carbonite from player 1 to player 2
                 context.player1.moveCard(context.evidenceOfTheCrime, 'hand');
                 context.player1.clickCard(context.evidenceOfTheCrime);
-                expect(context.player1).toBeAbleToSelectExactly([context.frozenInCarbonite, context.p1BobaFettArmor, context.p2BobaFettArmor]);
+                expect(context.player1).toBeAbleToSelectExactly([
+                    context.frozenInCarbonite,
+                    context.p1BobaFettArmor,
+                    context.p2BobaFettArmor,
+                    context.smugglingCompartment
+                ]);
 
                 context.player1.clickCard(context.frozenInCarbonite);
-                expect(context.player1).toBeAbleToSelectExactly([context.bobaFettDisintegrator, context.isbAgent, context.avenger, context.atst, context.cartelSpacer]);
+                expect(context.player1).toBeAbleToSelectExactly([
+                    context.bobaFettDisintegrator,
+                    context.isbAgent,
+                    context.avenger,
+                    context.greenSquadronAwing,
+                    context.atst,
+                    context.frontierAtrt,
+                    context.cartelSpacer
+                ]);
 
                 context.player1.clickCard(context.atst);
                 expect(context.avenger).toHaveExactUpgradeNames([]);
@@ -61,7 +86,12 @@ describe('Evidence of the Crime', function() {
                 const { context } = contextRef;
 
                 context.player1.clickCard(context.evidenceOfTheCrime);
-                expect(context.player1).toBeAbleToSelectExactly([context.frozenInCarbonite, context.p1BobaFettArmor, context.p2BobaFettArmor]);
+                expect(context.player1).toBeAbleToSelectExactly([
+                    context.frozenInCarbonite,
+                    context.p1BobaFettArmor,
+                    context.p2BobaFettArmor,
+                    context.smugglingCompartment
+                ]);
 
                 context.player1.clickCard(context.p1BobaFettArmor);
                 expect(context.player1).toBeAbleToSelectExactly([context.bobaFettDisintegrator, context.bobaFettDaimyo, context.isbAgent, context.darthVader]);
@@ -77,7 +107,12 @@ describe('Evidence of the Crime', function() {
                 const { context } = contextRef;
 
                 context.player1.clickCard(context.evidenceOfTheCrime);
-                expect(context.player1).toBeAbleToSelectExactly([context.frozenInCarbonite, context.p1BobaFettArmor, context.p2BobaFettArmor]);
+                expect(context.player1).toBeAbleToSelectExactly([
+                    context.frozenInCarbonite,
+                    context.p1BobaFettArmor,
+                    context.p2BobaFettArmor,
+                    context.smugglingCompartment
+                ]);
 
                 context.player1.clickCard(context.p2BobaFettArmor);
                 expect(context.player1).toBeAbleToSelectExactly([context.bobaFettDisintegrator, context.bobaFettDaimyo, context.isbAgent, context.darthVader]);
@@ -92,6 +127,32 @@ describe('Evidence of the Crime', function() {
                 expect(context.bobaFettDaimyo).toHaveExactUpgradeNames(['grievouss-wheel-bike']);
 
                 context.player2.passAction();
+            });
+
+            it('should be able to move enemy upgrades to friendly units in another arena', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.evidenceOfTheCrime);
+                expect(context.player1).toBeAbleToSelectExactly([
+                    context.frozenInCarbonite,
+                    context.p1BobaFettArmor,
+                    context.p2BobaFettArmor,
+                    context.smugglingCompartment
+                ]);
+
+                context.player1.clickCard(context.smugglingCompartment);
+                expect(context.player1).toBeAbleToSelectExactly([
+                    context.avenger,
+                    context.greenSquadronAwing,
+                    context.atst,
+                    context.frontierAtrt,
+                    context.cartelSpacer
+                ]);
+
+                context.player1.clickCard(context.greenSquadronAwing);
+
+                expect(context.greenSquadronAwing).toHaveExactUpgradeNames(['smuggling-compartment']);
+                expect(context.frontierAtrt).toHaveExactUpgradeNames([]);
             });
         });
     });

--- a/test/server/cards/04_JTL/units/ShuttleST149UnderKrennicsAuthority.spec.ts
+++ b/test/server/cards/04_JTL/units/ShuttleST149UnderKrennicsAuthority.spec.ts
@@ -40,10 +40,10 @@ describe('Shuttle ST-149 Under Krennics Authority', function() {
                     context.finn
                 ]);
 
-                context.player1.clickCard(context.fugitiveWookiee);
+                context.player1.clickCard(context.avenger);
 
                 expect(context.player2).toBeActivePlayer();
-                expect(context.fugitiveWookiee).toHaveExactUpgradeNames(['experience']);
+                expect(context.avenger).toHaveExactUpgradeNames(['shield', 'experience']);
 
                 // Shielded should resolve now
                 expect(context.shuttleSt149).toHaveExactUpgradeNames(['shield']);
@@ -61,7 +61,7 @@ describe('Shuttle ST-149 Under Krennics Authority', function() {
                     context.frozenInCarbonite
                 ]);
                 context.player1.clickCard(context.experience);
-                expect(context.fugitiveWookiee).toHaveExactUpgradeNames(['shield']);
+                expect(context.avenger).toHaveExactUpgradeNames(['shield', 'shield']);
             });
 
             it('should allow, when defeated, to move a token upgrade to another eligible unit with a different controller', function () {

